transport/http2_client.go

```diff
@@ -380,7 +380,11 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 	for _, c := range t.creds {
 		data, err := c.GetRequestMetadata(ctx, audience)
 		if err != nil {
-			return nil, streamErrorf(codes.Internal, "transport: %v", err)
+			if _, ok := err.(StreamError); ok {
```

Comment:
This should be checking whether the error was produced by the `status` package instead:

```go
if _, ok := status.FromError(err); ok {
  return nil, err
}
```

It's fine to keep returning `streamErrorf(codes.Unauthenticated)` below to avoid changing too much at once.

We also should update the comments in `credentials/credentials.go` for `GetRequestMetadata` to explain that if a `status` error is returned, it will be used as the status for the RPC.

Reply:
@dfawley 👍  done and done; I'll take a closer look at the error structs, apologies for that